### PR TITLE
Change json serialization behavior to ignore null [1189]

### PR DESCRIPTION
**Pull request by MarkZuber**
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1189

### CODE REVIEW

File: src/Microsoft.Identity.Client/Cache/Items/MsalAccountCacheItem.cs [hunk 1]
```diff
@@ -126,14 +126,16 @@ internal override JObject ToJObject()
...
+            SetItemIfValueNotNull(json, StorageJsonKeys.FamilyName, FamilyName);
+            SetItemIfValueNotNull(json, StorageJsonKeys.LocalAccountId, LocalAccountId);
+            SetItemIfValueNotNull(json, StorageJsonKeys.AuthorityType, AuthorityType);
+            SetItemIfValueNotNull(json, StorageJsonKeys.Realm, TenantId);
+
             // todo(cache): we don't support middle name json[StorageJsonKeys.MiddleName] = MiddleName;
-            json[StorageJsonKeys.LocalAccountId] = LocalAccountId;
-            json[StorageJsonKeys.AuthorityType] = AuthorityType;
-            json[StorageJsonKeys.Realm] = TenantId;
+            // SetItemIfValueNotNull(json, StorageJsonKeys.MiddleName, MiddleName);
```
Comment (jennyf19) [inline src/Microsoft.Identity.Client/Cache/Items/MsalAccountCacheItem.cs]: can we delete this?

File: src/Microsoft.Identity.Client/Cache/Items/CacheSerializationContract.cs [hunk 2]
```diff
@@ -180,7 +181,13 @@ internal string ToJsonString()
                 root[kvp.Key] = kvp.Value;
             }
 
-            return root.ToString();
+            return JsonConvert.SerializeObject(
+                root,
+                Formatting.Indented,
```
Comment (henrik-me) [inline src/Microsoft.Identity.Client/Cache/Items/CacheSerializationContract.cs]: >Formatting.Indented [](start = 16, length = 19)

why? don't we want as small size as possible?

File: src/Microsoft.Identity.Client/Cache/Items/MsalItemWithAdditionalFields.cs [hunk 3]
```diff
@@ -27,5 +26,30 @@ internal virtual JObject ToJObject()
 
             return json;
         }
+
+        internal void SetItemIfValueNotNull(JObject json, string key, JToken value)
+        {
+            bool shouldSetValue = true;
+
+            object asObj = value.ToObject<object>();
```
Comment (henrik-me) [inline src/Microsoft.Identity.Client/Cache/Items/MsalItemWithAdditionalFields.cs]: what will this do to int/numeric fields?

Comment (MarkZuber) [reply src/Microsoft.Identity.Client/Cache/Items/MsalItemWithAdditionalFields.cs]: It boxes the int and we get the numeric value in asObj so this works fine.

**Pull request merged**